Sources/SafeDICore/Models/TypeDescription.swift

```diff
@@ -425,6 +429,38 @@ extension ExprSyntax {
     }
 }
 
+// MARK: - UnorderedComparingArray
+
+public struct UnorderedComparingArray<Element: Codable & Hashable & Sendable>: Codable, Hashable, Sendable, Collection {
+
+    init(_ array: [Element]) {
+        self.array = array
+        set = Set(array)
+    }
+
+    let array: [Element]
+    private let set: Set<Element>
+
+    public static func == (lhs: UnorderedComparingArray, rhs: UnorderedComparingArray) -> Bool {
+        lhs.set == rhs.set
```

Comment:
Duh, easier than I thought

Reply:
Hardest part was coming up with a name for the type. I'm not convinced I did a great job there.